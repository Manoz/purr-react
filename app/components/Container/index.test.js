import React from 'react';
import { shallow } from 'enzyme';

import Container from './index';

describe('<Container />', () => {
  it('should render a <div>', () => {
    const wrapper = shallow(<Container />);
    expect(wrapper.type()).toEqual('div');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = shallow(<Container id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = shallow(<Container attribute="test" />);
    expect(wrapper.prop('attribute')).toBeUndefined();
  });
});
