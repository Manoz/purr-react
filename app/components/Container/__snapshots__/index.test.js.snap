// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render a <div> 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Container-o1cmc9-0",
        "isStatic": true,
        "rules": Array [
          "width:100%;padding:0 15px;margin:0 auto;@media (min-width:576px){max-width:540px;}@media (min-width:768px){max-width:720px;}@media (min-width:992px){max-width:960px;}@media (min-width:1200px){max-width:1140px;}",
        ],
      },
      "displayName": "Container",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Container-o1cmc9-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
